bindings/python/pmix.pxi

```diff
@@ -903,6 +903,57 @@ cdef void pmix_free_pdata(pmix_pdata_t *array, size_t sz):
         n += 1
     PyMem_Free(array)
 
+# Convert a dictionary of key-value pairs into an
+# array of pmix_query_t structs
+#
+# @array [INPUT]
+#        - malloc'd array of pmix_query_t structs
+#
+# @pyq [INPUT]
+#          - a list of dictionaries, where each
+#            dictionary has a list of keys, and a
+#            list of qualifers
+#
+cdef int pmix_load_queries(pmix_query_t *array, pyq:list):
+    n = 0
+    for q in pyq:
+        nstrings = len(q['keys'])
+        if 0 < nstrings:
+            array[n].keys = <char **> PyMem_Malloc(nstrings * sizeof(char*))
+            if not array[n].keys:
+                return PMIX_ERR_NOMEM
+        # load keys from dictionary
+        n = 0
```

Comment:
Looks like you are reusing 'n' inside of a loop that is indexing 'n' - so I suspect the routine is overwriting array[n]? I think you need a second counter here.